lib/sdk.dart

```diff
@@ -292,6 +290,15 @@ class SdkDiscovery {
     }
   }
 
+  /// Return the path to the Dart sdk contained within the Flutter sdk, if
+  /// either exist.
+  String _tryFlutterSdk() {
+    if (!_flutterSdkManager.hasSdk) return null;
+    return _flutterSdkManager.sdk.dartSdkPath;
+  }
+
+  FlutterSdkManager get _flutterSdkManager => deps[FlutterSdkManager];
+
   String _resolveSdkFromVm(String vmPath) {
     if (vmPath == null) return null;
```

Comment:
Does ```which``` return ```null``` or empty string if the path cannot be found?
Do we need to check for empty string as well as ```null```?